quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/VaultPropertyMappers.java

```diff
@@ -8,26 +8,45 @@ private VaultPropertyMappers() {
     public static PropertyMapper[] getVaultPropertyMappers() {
         return new PropertyMapper[] {
                 builder()
-                        .from("vault-file-path")
-                        .to("kc.spi-vault-files-plaintext-dir")
-                        .description("If set, secrets can be obtained by reading the content of files within the given path.")
+                        .from("vault")
+                        .description("Enables a vault provider.")
+                        .expectedValues("file", "hashicorp")
+                        .paramLabel("provider")
+                        .build(),
+                builder(false)
+                        .from("vault-dir")
+                        .to("kc.spi-vault-file-dir")
+                        .description("If set, secrets can be obtained by reading the content of files within the given directory.")
                         .paramLabel("dir")
                         .build(),
                 builder()
-                        .from("vault-hashicorp-")
+                        .from("vault-")
                         .to("quarkus.vault.")
-                        .description("If set, secrets can be obtained from Hashicorp Vault.")
+                        .description("Maps any vault option to their corresponding properties in quarkus-vault extension.")
+                        .hidden(true)
                         .build(),
                 builder()
-                        .from("vault-hashicorp-paths")
+                        .from("vault-url")
+                        .to("quarkus.vault.url")
+                        .description("The vault server url.")
+                        .paramLabel("paths")
+                        .hidden(true)
+                        .build(),
+                builder(false)
+                        .from("vault-kv-paths")
                         .to("kc.spi-vault-hashicorp-paths")
-                        .description("A set of one or more paths that should be used when looking up secrets.")
+                        .description("A set of one or more key/value paths that should be used when looking up secrets.")
                         .paramLabel("paths")
+                        .hidden(true)
                         .build()
         };
     }
 
     private static PropertyMapper.Builder builder() {
-        return PropertyMapper.builder(ConfigCategory.VAULT).isBuildTimeProperty(true);
+        return builder(true);
+    }
+
+    private static PropertyMapper.Builder builder(boolean buildTime) {
+        return PropertyMapper.builder(ConfigCategory.VAULT).isBuildTimeProperty(buildTime);
```

Comment:
hum, can't we do it the same way as everywhere else in the mappers, by just adding this to the property? just a slight bit more readable imo, i can see if the property is buildTime or not at first sight, so not really hard on this, though. See e.g. https://github.com/keycloak/keycloak/blob/438fc2865f019cebaa1e1fea762a1f6c77038b7b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HttpPropertyMappers.java#L36-L42 - no need to scroll to see what that "true" or "false" actually means.

Reply:
Makes sense.